### PR TITLE
Compute v2: Generate a keypair when no public key is specified

### LINKED ISSUE #210: openstack_compute_keypair_v2.pki: diffs didn't match during apply
I was trying to do automated provisioning of ssh keypair in CI/CD pipeline and ended up with this. The problem is on my side (one should not create keypairs with empty `public_key`), but Terraform politely asked me to create Issue, and you know - it's Christmas, so wishes should be fulfilled.

So here it is.

### Terraform Version

0.11.0

* provider.openstack: version = "~> 1.1"

### Affected Resource(s)
- openstack_compute_keypair_v2

### Terraform Configuration Files
```hcl
resource "openstack_compute_keypair_v2" "pki" {
  name       = "pki-test"
  public_key = ""
}
```

### Panic Output

```
$ terraform-0.11.0 apply -auto-approve
openstack_compute_keypair_v2.pki: Creating...
  name:   "" => "pki-test"
  region: "" => "<computed>"
openstack_compute_keypair_v2.pki: Creation complete after 1s (ID: pki-test)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
ubuntu@deployer:~/devel/pki-boulder/provision/2$ terraform-0.11.0 apply -auto-approve
openstack_compute_keypair_v2.pki: Refreshing state... (ID: pki-test)
openstack_compute_keypair_v2.pki: Destroying... (ID: pki-test)
openstack_compute_keypair_v2.pki: Destruction complete after 0s

Error: Error applying plan:

1 error(s) occurred:

* openstack_compute_keypair_v2.pki: openstack_compute_keypair_v2.pki: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.11.0
    Resource ID: openstack_compute_keypair_v2.pki
    Mismatch reason: attribute mismatch: public_key
    Diff One (usually from plan): *terraform.InstanceDiff{mu:sync.Mutex{state:0, sema:0x0}, Attributes:map[string]*terraform.ResourceAttrDiff{"name":*terraform.ResourceAttrDiff{Old:"pki-test", New:"pki-test", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "public_key":*terraform.ResourceAttrDiff{Old:"ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDYKMPxnlSgis2lCO+ZT1liiI/vUNYUU+u5ipU9wWNdL/hJdELoySJYORBBcWLZpuerKVe5cZu0TgWZxUwPLPjn6WaHGawV2I1GvHxQ/+H0aFnHKkJzbTLGfs3EeHEbUJRm5ZXFYkyYCNPfgM/4MFPVWX6W1j/Z0ZzY+Zn5SeU1GsahVyRTf9w4ywva/H4fSNzfmjgRucM+AXO+Cf60hX5fprYKFvF4LMucUUdQ7hk94hv+EFMOCsIQ6w2J+vWz0+zyY4ISoBbyns6ash0a1qufjKFtWasFpPZVplx6yxPWlCWjlBSlg+GYDoG48nAPzEYIM3Op4iJzcVGMy+itiel5 Generated-by-Nova", New:"", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Sensitive:false, Type:0x0}, "region":*terraform.ResourceAttrDiff{Old:"RegionOne", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}}, Destroy:false, DestroyDeposed:false, DestroyTainted:false, Meta:map[string]interface {}(nil)}
    Diff Two (usually from apply): *terraform.InstanceDiff{mu:sync.Mutex{state:0, sema:0x0}, Attributes:map[string]*terraform.ResourceAttrDiff{"name":*terraform.ResourceAttrDiff{Old:"", New:"pki-test", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Sensitive:false, Type:0x0}, "region":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Sensitive:false, Type:0x0}}, Destroy:false, DestroyDeposed:false, DestroyTainted:false, Meta:map[string]interface {}(nil)}

Also include as much context as you can about your config, state, and the steps you performed to trigger this error.


Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

### Expected Behavior
It should probably complain about `ssh_key` attribute not present, or empty.

### Actual Behavior
See above

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### PR DESCRIPTION
**Pull request by jtopjian**
This commit will allow a public/private keypair to be generated
when no public key is specified.

Fixes #210

### CODE REVIEW

**Comment by jtopjian** [comment]
The security warning in the docs was taken from the TLS provider:

https://www.terraform.io/docs/providers/tls/r/private_key.html

File: website/docs/r/compute_keypair_v2.html.markdown [hunk 1]
```diff
@@ -10,15 +10,31 @@ description: |-
...
 
+~> **Important Security Notice** The private key generated by this resource will
+be stored *unencrypted* in your Terraform state file. **Use of this resource
+for production deployments is *not* recommended**. Instead, generate
+a private key file outside of Terraform and distribute it securely
+to the system where Terraform will be run.
+
 ## Example Usage
 
+### Import an Existing Private Key
```
Comment (fatmcgav) [inline website/docs/r/compute_keypair_v2.html.markdown]: `s/Private/Public`

Comment (jtopjian) [reply website/docs/r/compute_keypair_v2.html.markdown]: Good catch. Thank you!

**Comment by jtopjian** [comment]
@fatmcgav thanks!

**Pull request merged**